stl/inc/chrono

```diff
@@ -4613,8 +4613,10 @@ namespace chrono {
             const _Ctype& _Ctype_fac{_STD use_facet<_Ctype>(_Iosbase.getloc())};
             constexpr _InIt _Last{};
 
-            while (*_Fmt != '\0' && _First != _Last && _State == ios_base::goodbit) {
-                if (*_Fmt == '%') {
+            while (*_Fmt != '\0' && (_State & ~ios_base::eofbit) == ios_base::goodbit) {
+                if (_First == _Last) {
+                    _State |= ios_base::failbit;

```

Comment:
I think this should also set `eofbit`. Consider
```cpp
istringstream src("2021-"s);
year_month ym;
src >> parse("%Y-%m-%d", ym);
```
This will set `eofbit | failbit`. With this code, on the other hand,
```cpp
istringstream src("2021-"s);
year_month ym;
src >> parse("%F", ym);
```
only sets `failbit`. I think these would make good additional test cases, but you won't be able to use the helper function `fail_parse` because you need to test the exact `rdstate`.

```suggestion
                    _State |= ios_base::failbit | ios_base::eofbit;
```